modules/apps/frontend-view-state/frontend-view-state-service/src/main/java/com/liferay/frontend/view/state/internal/jaxrs/context/provider/UserContextProvider.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.view.state.internal.jaxrs.context.provider;
+
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.util.Portal;
+
+import javax.servlet.http.HttpServletRequest;
+
+import javax.ws.rs.ext.Provider;
+
+import org.apache.cxf.jaxrs.ext.ContextProvider;
+import org.apache.cxf.message.Message;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(immediate = true, service = UserContextProvider.class)
+@Provider
+public class UserContextProvider implements ContextProvider<User> {
```

Comment:
Portal Vulcan has already a UserContextProvider, in case you decide that it is worth to use the Portal Vulcan infrastructure